docs/content/guides/developer/getting-started/data-serving.mdx

```diff
@@ -0,0 +1,115 @@
+- Utilize a [RPC provider](https://sui.io/developers#dev-tools) (filter by `RPC`) or [Data indexer operator](https://github.com/sui-foundation/awesome-sui?tab=readme-ov-file#indexers--data-services) who have setup their own full nodes or indexing infrastructure.
+
+:::info
+
+You can use one of the [future-oriented interfaces]((#future-state-data-interfaces)) that are available in alpha or beta, but you must validate the relevant functionality, performance, and cost characteristics for your application before deciding to use those in production.
+
+:::
+
+### JSON-RPC
+
+You can currently get real-time or historical data from a Sui full node. Retention period for historical data depends on the [pruning strategy](/guides/operator/data-management.mdx#sui-full-node-pruning-policies) implemented by a node operator, though currently all full nodes are by default configured to implicitly fall back on a [Mysten Labs managed key-value store for historical transaction data](/guides/operator/data-management.mdx#sui-full-node-key-value-store-backup).
+
+:::caution
+
+Websockets based JSON RPCs `suix_subscribeEvent` and `suix_subscribeTransaction` were deprecated in July 2024. You should not rely on those RPCs in your applications. Refer to [this section](#future-state-data-interfaces) to learn about a future alternative.
+
+:::
+
+### Custom indexer
+
+If you need object version history or another type of historical data where you may not fully rely on a full node or RPC provider, or if you've specific query patterns that could be best served from a relational database, then you can setup your own [custom indexer](/guides/developer/advanced/custom-indexer.mdx) or reach out to a [Data indexer operator](https://github.com/sui-foundation/awesome-sui?tab=readme-ov-file#indexers--data-services) who may already have setup one. 
+
+If you setup your own indexer, you are responsible for its ongoing maintenance and the related infrastructure & operational costs. You can reduce your costs by implementing a pruning strategy for the relational database by taking into account the retention needs of your application.
+
+## Future data access interfaces
+
+Primary interfaces to access Sui access data in future will be:
+
+- Full node gRPC API will replace JSON-RPC on full nodes, and is currently available in alpha. If you already use the JSON-RPC or are starting to utilize it as a dependency for your use case, you would need to migrate to gRPC or GraphQL (see below) within a reasonable duration. Refer to the high-level timeline for gRPC availability [below](#grpc-api).
+- [GraphQL RPC with Indexer 2.0](/references/sui-graphql.mdx) will replace the custom indexer mechanism and is currently available in beta. This will be a combination of a performant indexing mechanism that one could use to load and manage data in a postgres relational database, and a lightweight GraphQL RPC server that will be used to read data from that database.
```

Comment:
> will replace the custom indexer mechanism and is currently available in beta

IIUC this isn't quite accurate. We want to continue supporting folks writing their own custom indexers or custom data processors utilizing the indexing framework. Graphql is one such product that leverages the indexing framework